File: backend/models/workflow/workflow_template.go
package workflow

import (
	"time"

	"gorm.io/gorm"
)

// WorkflowTemplate 工作流模板文件（可复用的配置模板，供 Stage Template 编排时选取）
type WorkflowTemplate struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	Name        string         `json:"name" gorm:"size:100;not null"`
	Description string         `json:"description" gorm:"size:500"`
	Content     string         `json:"content" gorm:"type:text"`   // Go template 语法内容
	Variables   string         `json:"variables" gorm:"type:text"` // 变量说明文档（JSON格式）
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

func (WorkflowTemplate) TableName() string { return "workflow_templates" }

File: frontend/src/api/workflowTemplate.ts
import request from '@/utils/request'

export interface WorkflowTemplate {
  id: number
  name: string
  description: string
  content: string
  variables: string
  created_at: string
  updated_at: string
}

/** 获取所有工作流模板文件 */
export function getWorkflowTemplatesApi(): Promise<WorkflowTemplate[]> {
  return request.get('/workflow-templates').then((res) => res.data.data)
}

/** 获取工作流模板文件详情 */
export function getWorkflowTemplateApi(id: number): Promise<WorkflowTemplate> {
  return request.get(`/workflow-templates/${id}`).then((res) => res.data.data)
}

/** 创建工作流模板文件 */
export function createWorkflowTemplateApi(data: Partial<WorkflowTemplate>): Promise<WorkflowTemplate> {
  return request.post('/workflow-templates', data).then((res) => res.data.data)
}

/** 更新工作流模板文件 */
export function updateWorkflowTemplateApi(id: number, data: Partial<WorkflowTemplate>): Promise<void> {
  return request.put(`/workflow-templates/${id}`, data).then((res) => res.data)
}

/** 删除工作流模板文件 */
export function deleteWorkflowTemplateApi(id: number): Promise<void> {
  return request.delete(`/workflow-templates/${id}`).then((res) => res.data)
}
